evm-contracts/test/v0.6/FluxAggregator.test.ts

```diff
@@ -1052,23 +1052,73 @@ describe('FluxAggregator', () => {
       })
     })
 
-    const limit = 42
+    const limit = 100
     describe(`when adding more than ${limit} oracles`, () => {
-      it('reverts', async () => {
+      let oracles: ethers.Wallet[]
+
+      beforeEach(async () => {
+        oracles = []
+        for (let i = 0; i < limit; i++) {
+          const account = await wallet.createWallet(provider, i + 100)
+          await personas.Default.sendTransaction({
+            to: account.address,
+            value: h.toWei('0.01'),
+          })
+          oracles.push(account)
+        }
+
         await link.transfer(
           aggregator.address,
           paymentAmount.mul(limit).mul(reserveRounds),
         )
         await aggregator.updateAvailableFunds()
 
-        for (let i = 0; i < limit; i++) {
-          const minMax = i + 1
-          const fakeAddress = h.addHexPrefix(randomBytes(20).toString('hex'))
+        let addresses = oracles.slice(0, 50).map(o => o.address)
+        await aggregator
+          .connect(personas.Carol)
+          .addOracles(addresses, addresses, 1, 50, rrDelay)
+        // add in two transactions to avoid gas limit issues
+        addresses = oracles.slice(50, 100).map(o => o.address)
+        await aggregator
+          .connect(personas.Carol)
+          .addOracles(addresses, addresses, 1, 100, rrDelay)
+      })
 
-          await aggregator
-            .connect(personas.Carol)
-            .addOracles([fakeAddress], [fakeAddress], minMax, minMax, rrDelay)
+      it.only('not use too much gas', async () => {
+        let tx: any
+        // const inputs = adverserialQuickselectList(limit)
+        for (let i = 0; i < limit; i++) {
+          tx = await aggregator
+            .connect(oracles[i])
+            .submit(nextRound, Math.floor(Math.random() * 100000))
+        }
+        assert(!!tx)
+        if (!!tx) {
+          const receipt = await tx.wait()
+          assert.isAbove(500_000, receipt.gasUsed.toNumber())
+          console.log('#############', receipt.gasUsed.toNumber())
         }
+      })
+
+      // function adverserialQuickselectList(len: number): number[] {
+      //   let xs: number[] = []
+      //   let pi: number[] = []
+      //   for (let i = 0; i < len; i++) {
+      //     pi[i] = i
+      //     xs[i] = 0
+      //   }
+
+      //   for (let l = len - 1; l > 0; l--) {
```

Comment:
```suggestion
      //   for (let l = len; l > 0; l--) {
```
There are only two hard things in computer science: cache invalidation, naming things, and off-by-one errors. :-)

Reply:
I also suggest doing an `assert.deepEqual` against a golden output, e.g. for `len == 16`, you should get `[2, 4, 6, 8, 10, 12, 14, 16, 1, 9, 5, 11, 3, 13, 7, 15]`